tests/integration/views/account/test_password_change.py

```diff
@@ -0,0 +1,130 @@
+        self.contact_details_form = {
+            "option": "contact_details"
+        }
+        self.patcher.start()
+        self.auth_error = {
+            'detail': 'Unauthorized user credentials'
+        }
+
+    def tearDown(self):
+        self.patcher.stop()
+
+    @requests_mock.mock()
+    def test_account_password_change_option_select(self, mock_request):
+        mock_request.get(url_banner_api, status_code=404)
+        mock_request.get(url_respondent_party_by_id, status_code=404)
+
+        response = self.app.post('/my-account', data={"option": 'change_password'}, follow_redirects=True)
+        self.assertEqual(response.status_code, 200)
+        self.assertTrue('Change your password'.encode() in response.data)
+        self.assertTrue('Enter your current password'.encode() in response.data)
+        self.assertTrue('Your password must have:'.encode() in response.data)
+        self.assertTrue('at least 8 characters'.encode() in response.data)
+        self.assertTrue('at least 1 uppercase letter'.encode() in response.data)
+        self.assertTrue('at least 1 symbol (eg: ?!£%)'.encode() in response.data)
+        self.assertTrue('at least 1 number'.encode() in response.data)
+        self.assertTrue('New Password'.encode() in response.data)
+        self.assertTrue('Re-type your new password'.encode() in response.data)
+
+    @requests_mock.mock()
+    @patch('frontstage.controllers.party_controller.get_respondent_party_by_id')
+    def test_account_password_change_success(self, mock_request, get_respondent_party_by_id):
+        mock_request.get(url_banner_api, status_code=404)
+        mock_request.get(url_respondent_party_by_id, status_code=404)
+        mock_request.post(url_auth_token, status_code=200)
+        mock_request.put(url_password_change, status_code=200)
+        get_respondent_party_by_id.return_value = respondent_party
+        response = self.app.post('/my-account/change-password', data={"password": 'test',
+                                                                      "new_password": 'Password123!',
+                                                                      "new_password_confirm": 'Password123!'},
+                                 follow_redirects=True)
+        self.assertEqual(response.status_code, 200)
+        self.assertTrue('Your password has been changed.'.encode() in response.data)
+        self.assertTrue('using your enrolment code'.encode() in response.data)
+
+    @requests_mock.mock()
+    @patch('frontstage.controllers.party_controller.get_respondent_party_by_id')
+    def test_account_password_change_form_validation_errors_does_not_meet_requirement(self, mock_request,
+                                                                                      get_respondent_party_by_id):
+        mock_request.get(url_banner_api, status_code=404)
+        mock_request.get(url_respondent_party_by_id, status_code=404)
```

Comment:
Same as above,  for this (and all the other tests) change the auth_token to 204 status (from 200), and change this to a 200 with data (from 404), even if you're not going to use the data.

Reply:
👍